.circleci/config.yml

```diff
@@ -253,62 +350,126 @@ jobs:
               docker stop $SERVICE_NAME && docker rm $SERVICE_NAME
               docker run --runtime=nvidia --name $SERVICE_NAME --restart unless-stopped -v $ETCD_VOLUME:$ETCD_CONTAINER_FOLDER -p $SNETD_PORT:$SNETD_PORT -di $DOCKER_ORG/$SERVICE_NAME:$CIRCLE_SHA1 python3.6 run_service.py --daemon-config-path $SNETD_CONFIG
             EOF
+
   build-sentiment-analysis-container:
-    working_directory: /sentiment-analysis
-    docker:
-      - image: docker:18.06.1-ce-git
-    environment:
-      SERVICE_NAME: sentiment-analysis
-      ORG: singnet
-      DOCKER_ORG: singularitynet
-    <<: *default_steps_br
-  deploy-sentiment-analysis:
     machine:
-      enabled: true
+      image: circleci/classic:201808-01
+    working_directory: ~/singnet
     environment:
       SERVICE_NAME: sentiment-analysis
-      ORG: singnet
-      DOCKER_ORG: singularitynet
-      SNETD_PORT: 7010
-      SNETD_CONFIG: config/snetd_sentiment_analysis_server_config.json
+      CACHE_MODELS_PATH: sentiment-analysis/caches/models
     steps:
+      - checkout
+      - build_service:
+          service_name: sentiment-analysis
+          org: singnet
+          docker_org: singularitynet
       - run:
-          name: Deploy Over SSH
+          name: Install NLTK depedencies.
           command: |
-            ssh $SSH_USER_BR@$SSH_HOST_BR \<< EOF
-              docker login -u $DOCKER_HUB_USER_ID -p $DOCKER_HUB_PWD;
-              docker pull $DOCKER_ORG/$SERVICE_NAME:$CIRCLE_SHA1
-              docker stop $SERVICE_NAME && docker rm $SERVICE_NAME
-              docker run --runtime=nvidia --name $SERVICE_NAME --restart unless-stopped -p $SNETD_PORT:$SNETD_PORT -v /home/ubuntu/nlp-services/$SERVICE_NAME/config:/$SERVICE_NAME/config -di $DOCKER_ORG/$SERVICE_NAME:$CIRCLE_SHA1 python3.6 run_service.py --daemon-config-path $SNETD_CONFIG
-            EOF
-  build-named-entity-recognition-container:
-    working_directory: /named-entity-recognition
-    docker:
-      - image: docker:18.06.1-ce-git
-    environment:
-      SERVICE_NAME: named-entity-recognition
-      ORG: singnet
-      DOCKER_ORG: singularitynet
-    <<: *default_steps_br
-  deploy-named-entity-recognition:
+            cd $SERVICE_NAME
```

Comment:
It looks like you build the docker image, but then you do all the testing on the circleci host? (As well as downloading the models)

It seems like it'd make more sense to do these tests first, then only build and push a docker image after the tests are successful.

You could also reuse the models and pip cache that you download on circleci by adding them to the container as part of the docker context.

Lastly, the `build_service` command currently mixes building a pyenv on the circleci host, with building the docker image. These should two separate commands.

Reply:
Another option would be to use the docker image to run the tests, because as it is, circleci will be spending time to set up dependencies for two environments... the docker image *and* the circleci host.

Reply:
We have turned the "Build forked pull requests" feature off on CircleCI,  all the status checks on open PR comes from my personal CircleCI builds, my personal CircleCI can't build your services because you are using singnet dockerhub credentials to push images. My services use my personal dockerhub credentials to push my development image versions, if the PR gets merged, it will use singnet credentials.

_* Does this mean every PR gets deployed?_
No, all builds and deployments are happening in the development environment side (developer environment). As a developer, I also have a development environment to deploy my development versions.

_* You could also reuse the models and pip cache that you download on circleci by adding them to the container as part of the docker context._
Before we already had a cache of dependencies on circleci and docker image and we were not installing dependencies twice but, once we are already cache the docker image with dependencies, I've removed the download of dependencies on circleci. 

Now we are using docker image to run the tests.

About promoting a staging deployment to production, we are discussing about it in Brazil. I think we can discuss this question better on slack.

Reply:
Ok, well we can't some some services broken in PR builds. I'm happy for things to be changed, but if my services are broken, then they also need to be included in the proposal. It's not acceptable to break other people's work and leave it to them to clean up.

Re: PR deployments. You said "No", but then said they get deployed to a development environment :confused:  If you open two pull requests, how will you know which one has been deployed to your development environment?

Reply:
For sure, it is not acceptable to break other people's work, for that reason, I told you on slack, this is a proposal and before I start changing other's people work, I need to know if I can change  their work. Once you agreed with these changes, I can do this without problems.

About PR deployments I said:
_As a developer, I also have a development environment to deploy my development versions._ 
It means my personal development environment, not SingularityNET's development environment.
If I open a PR, it will be deployed only on my personal(Glauter) development environment, I mean, CircleCI, dockerhub, aws or any server where I can host my service. The same for the second, third, etc, open pull request. Each PR will be deployed on the pull request owner environment.

Once a pull request gets merged to master, then it will be deployed to SingularityNET production server or somewhere we decide to deploy it.